plumbing/format/packfile/scanner.go

```diff
@@ -275,19 +276,28 @@ func (s *Scanner) NextObject(w io.Writer) (written int64, crc32 uint32, err erro
 // ReadRegularObject reads and write a non-deltified object
 // from it zlib stream in an object entry in the packfile.
 func (s *Scanner) copyObject(w io.Writer) (int64, error) {
-	zr, err := zlib.NewReader(s.r)
-	if err != nil {
-		return -1, fmt.Errorf("zlib reading error: %s", err)
+	var err error
+	if s.zr == nil {
+		zr, err := zlib.NewReader(s.r)
+		if err != nil {
+			return -1, fmt.Errorf("zlib initialization error: %s", err)
```

Comment:
Why are we returning -1, this is not C!

`Scanner.NextObject` first return argument is not used anywhere, it is safe to return 0 here instead.